internal/provider/models/alerts/test/threshold_test.go

```diff
@@ -0,0 +1,728 @@
+							expressions = [
+								{
+									field = ".event_count"
+									operator = "greater"
+									value_number = 10
+								}
+							],
+						}
+						subject = "Threshold Alert for Log event"
+						message = "You received a threshold alert for a Log event"
+						ingestion_key = "abc123"
+					}`,
+				ExpectError: regexp.MustCompile("(?s).*Attribute component_id string length must be at least 1"),
+			},
+			// invalid component_kind
+			{
+				Config: GetCachedConfig(cacheKey) + `
+					resource "mezmo_threshold_alert" "bad_component_kind" {
+						pipeline_id = mezmo_pipeline.test_parent.id
+						component_kind = "NOPE"
+						component_id = mezmo_http_source.my_source.id
+						inputs = [mezmo_http_source.my_source.id]
+						name = "my threshold alert"
+						event_type = "metric"
+						operation = "sum"
+						conditional = {
+							expressions = [
+								{
+									field = ".event_count"
+									operator = "greater"
+									value_number = 10
+								}
+							],
+						}
+						subject = "Threshold Alert for Log event"
+						message = "You received a threshold alert for a Log event"
+						ingestion_key = "abc123"
+					}`,
+				ExpectError: regexp.MustCompile(`(?s).*Attribute component_kind value must be one of: \["source" "transform"\]`),
+			},
+			// Missing inputs
+			{
+				Config: GetCachedConfig(cacheKey) + `
+					resource "mezmo_threshold_alert" "missing_inputs" {
+						pipeline_id = mezmo_pipeline.test_parent.id
+						component_kind = "source"
+						component_id = mezmo_http_source.my_source.id
+						inputs = []
+						name = "my threshold alert"
+						event_type = "metric"
+						operation = "sum"
+						conditional = {
+							expressions = [
+								{
+									field = ".event_count"
+									operator = "greater"
+									value_number = 10
+								}
+							],
+						}
+						subject = "Threshold Alert for Log event"
+						message = "You received a threshold alert for a Log event"
+						ingestion_key = "abc123"
+					}`,
+				ExpectError: regexp.MustCompile(`(?s).*Attribute inputs list must contain at least 1 elements`),
+			},
+			// Inputs have bad length
+			{
+				Config: GetCachedConfig(cacheKey) + `
+					resource "mezmo_threshold_alert" "blank_inputs" {
+						pipeline_id = mezmo_pipeline.test_parent.id
+						component_kind = "source"
+						component_id = mezmo_http_source.my_source.id
+						inputs = [""]
+						name = "my threshold alert"
+						event_type = "metric"
+						operation = "sum"
+						conditional = {
+							expressions = [
+								{
+									field = ".event_count"
+									operator = "greater"
+									value_number = 10
+								}
+							],
+						}
+						subject = "Threshold Alert for Log event"
+						message = "You received a threshold alert for a Log event"
+						ingestion_key = "abc123"
+					}`,
+				ExpectError: regexp.MustCompile(`(?s).*Attribute inputs\[0\] string length must be at least 1`),
+			},
+			// // Inputs have duplicates
```

Comment:
Can this be uncommented or removed?

Reply:
We *can*, but I was hoping one of us would eventually solve the problem.  If this test is uncommented, it errors during teardown and I have no idea why.

If it's bothersome, I can remove, or add a comment to explain why it's there.

Reply:
Makes sense. I just wanted to make sure it was intentional and not an oversight.

Reply:
commented.